src/catalog/src/memory/objects.rs

```diff
@@ -1104,6 +1264,53 @@ impl CatalogItem {
         }
     }
 
+    /// Returns the [`GlobalId`]s that reference this item, if any.
+    pub fn global_ids(&self) -> impl Iterator<Item = GlobalId> + '_ {
+        let gid = match self {
+            CatalogItem::Source(source) => source.global_id,
+            CatalogItem::Log(log) => log.global_id,
+            CatalogItem::Sink(sink) => sink.global_id,
+            CatalogItem::View(view) => view.global_id,
+            CatalogItem::MaterializedView(mv) => mv.global_id,
+            CatalogItem::ContinualTask(ct) => ct.global_id,
+            CatalogItem::Index(index) => index.global_id,
+            CatalogItem::Func(func) => func.global_id,
+            CatalogItem::Type(ty) => ty.global_id,
+            CatalogItem::Secret(secret) => secret.global_id,
+            CatalogItem::Connection(conn) => conn.global_id,
+            CatalogItem::Table(table) => {
+                return itertools::Either::Left(table.collections.values().copied());
+            }
+        };
+        itertools::Either::Right(std::iter::once(gid))
```

Comment:
These are a way to return different kinds of iterators without `Box`-ing them. This method returns `impl Iterator` and Rust needs to resolve that to a single type, in this case the type is `itertools::Either` instead of `Box<dyn Iterator>`